lib/rbs/definition_builder.rb

```diff
@@ -86,6 +86,19 @@ def define_instance(definition, type_name, subst)
       one_ancestors = ancestor_builder.one_instance_ancestors(type_name)
       methods = method_builder.build_instance(type_name)
 
+      self_type_methods = one_ancestors.each_self_type.with_object({}) do |self_type, hash| #$ Hash[Symbol, Definition::Method]
```

Comment:
We cannot simply inject the `self_types` methods to the result `methods`/`definition`, because the `self_types` methods may include other inherited methods, like `#initialize` and having them in the module is something unexpected.